sklearn/cross_decomposition/pls_.py

```diff
@@ -367,6 +370,11 @@ def fit(self, X, Y):
             # => B = W*Q' (p x q)
             self.coef_ = np.dot(self.x_rotations_, self.y_loadings_.T)
             self.coef_ = self.coef_ * self.y_std_
+
+            if Y_ndim_is_one:
+                self.coef_ = np.ravel(self.coef_)
```

Comment:
Yes, from what I understand, LinearRegression learns the shape of the predict() output in the fit() function : see [here](https://github.com/scikit-learn/scikit-learn/blob/c393bd499ed51cdcf2e80e4c2fc9552c4e582858/sklearn/linear_model/base.py#L508-L509).

I think in its first state, _PLS was only working with 2D arrays. It was fixed by 82945e5. But reshaping the output of _PLS.predict was still missing and it is fixed by this PR